app/controllers/plans_controller.rb

```diff
@@ -56,4 +66,8 @@ def set_plans
   def set_all_features
     @all_features = Feature.all.collect { |x| [x.name, x.id] }
   end
+
+  def set_subscription_plan
+    @plan = Plan.find(params[:plan_id])
```

Comment:
by the way, can't we use already implemented set_plan method for this?

Reply:
Because in set plan we get attribute of `id` since we are using table we are getting attribute `plan_id`.

So we cannot use same method.

Reply:
yes, we can.
`@plan = Plan.find_by_id(params[:id] || params[:plan_id])`
Just make sure that in second case, we are not receiving `id` parameter. Otherwise they will both conflict with each other. But if you are only getting `plan_id` then it will work.